explorer/interpreter/interpreter.cpp

```diff
@@ -1087,7 +1085,23 @@ auto Interpreter::StepExp() -> ErrorOr<Success> {
       if (auto rewrite = op.rewritten_form()) {
         return todo_.ReplaceWith(std::make_unique<ExpressionAction>(*rewrite));
       }
-      if (act.pos() != static_cast<int>(op.arguments().size())) {
+      if ((op.op() == Operator::And || op.op() == Operator::Or) &&
+          act.pos() >= 1) {
```

Comment:
Missing a state representation if we want to be consistent, one is going finish, and the other to pop an action